lib/cli/src/project_types.ts

```diff
@@ -103,6 +105,15 @@ export const supportedTemplates: TemplateConfiguration[] = [
       return files.every(Boolean);
     },
   },
+  {
+    preset: ProjectType.VUE3,
+    dependencies: {
+      vue: (v) => v.replace(/^[^\w]+/, '').charAt(0) === '3',
```

Comment:
@SasanFarrokh would it be cleaner to use the semver library for this?

Reply:
Great idea